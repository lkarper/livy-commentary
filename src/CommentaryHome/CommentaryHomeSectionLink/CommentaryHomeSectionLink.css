.CommentaryHomeSectionLink__a {
    color: black;
    text-decoration: none;
}

.CommentaryHomeSectionLink__a:hover, .CommentaryHomeSectionLink__a:focus {
    text-decoration: underline;
}