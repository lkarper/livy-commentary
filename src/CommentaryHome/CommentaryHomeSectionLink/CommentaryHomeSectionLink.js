import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const CommentaryHomeSectionLink = (props) => {
    const { section } = props;

    return (
        <li>
            <Link 
                to={`/commentary-read/${section.section_number}`}
            >
                ยง{section.section_number}
            </Link>
        </li>
    );
}

CommentaryHomeSectionLink.defaultProps = {
    section: {
        section_number: 0,
    },
};

CommentaryHomeSectionLink.propTypes = {
    section: PropTypes.shape({
        section_number: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    }),
};

export default CommentaryHomeSectionLink;