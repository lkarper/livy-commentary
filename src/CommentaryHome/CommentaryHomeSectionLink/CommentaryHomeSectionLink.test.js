import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import CommentaryHomeSectionLink from './CommentaryHomeSectionLink';

describe('CommentaryHomeSectionLink component', () => {
    it('renders the UI as expected', () => {
        const div = document.createElement('div');
        ReactDOM.render(
            <BrowserRouter>
                <CommentaryHomeSectionLink />
            </BrowserRouter>,
            div
        );
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(
            <BrowserRouter>
                <CommentaryHomeSectionLink />
            </BrowserRouter>
        );
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
