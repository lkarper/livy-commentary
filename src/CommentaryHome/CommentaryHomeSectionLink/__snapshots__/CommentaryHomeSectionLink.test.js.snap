// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentaryHomeSectionLink component renders the UI as expected 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CommentaryHomeSectionLink
      section={
        Object {
          "section_number": 0,
        }
      }
    >
      <li>
        <Link
          to="/commentary-read/0"
        >
          <LinkAnchor
            href="/commentary-read/0"
            navigate={[Function]}
          >
            <a
              href="/commentary-read/0"
              onClick={[Function]}
            >
              ยง
              0
            </a>
          </LinkAnchor>
        </Link>
      </li>
    </CommentaryHomeSectionLink>
  </Router>
</BrowserRouter>
`;
